pandas/core/generic.py

```diff
@@ -1303,8 +1303,81 @@ def __invert__(self):
 
     def equals(self, other):
         """
-        Determines if two NDFrame objects contain the same elements. NaNs in
-        the same location are considered equal.
+        Test whether two NDFrame objects contain the same elements.
+
+        This function allows two NDFrame objects to be compared against
+        each other to see if they have the same shape and elements. NaNs in
+        the same location are considered equal. The column headers do not
+        need to have the same type, but the elements within the columns must
+        be the same dtype.
+
+        Parameters
+        ----------
+        other : NDFrame
+            The other NDFrame to be compared with the first.
+
+        Returns
+        -------
+        bool
+            True if all elements are the same in both NDFrames, False
+            otherwise.
+
+        See Also
+        --------
+        Series.eq : Compare two Series objects of the same length
+            and return a Series where each element is True if the element
+            in each Series is equal, False otherwise.
+        DataFrame.eq : Compare two DataFrame objects of the same shape and
+            return a DataFrame where each element is True if the respective
+            element in each DataFrame is equal, False otherwise.
+        numpy.array_equal : Return True if two arrays have the same shape
+            and elements, False otherwise.
```

Comment:
We add the method being documented to the `See Also`, only in cases where the docstring is being reused by many methods, and we want those methods to reference each other. But this is not the case here. May be you can still add the `pandas.utils.testing.assert_frame_equal` and `assert_series_equal` here, but I don't think it adds value to reference `.equals` itself.